xbmc/addons/AddonManager.cpp

```diff
@@ -479,7 +449,61 @@ bool CAddonMgr::HasAvailableUpdates()
   return !GetAvailableUpdates().empty();
 }
 
-bool CAddonMgr::GetAddons(const TYPE &type, VECADDONS &addons, bool enabled /* = true */)
+bool CAddonMgr::GetAddons(VECADDONS& addons)
+{
+  return GetAddonsInternal(ADDON_UNKNOWN, addons, true);
+}
+
+bool CAddonMgr::GetAddons(VECADDONS& addons, const TYPE& type)
+{
+  return GetAddonsInternal(type, addons, true);
+}
+
+bool CAddonMgr::GetInstalledAddons(VECADDONS& addons)
+{
+  return GetAddonsInternal(ADDON_UNKNOWN, addons, false);
+}
+
+bool CAddonMgr::GetInstalledAddons(VECADDONS& addons, const TYPE& type)
```

Comment:
aren't there any downsides of overloads, like increasing memory usage because of bigger objects etc?
And I think the biggest issues here currently is that `ADDON_UNKNOWN` is a bad naming for a type. `ADDON_TYPE_ANY` would be more clear and then also not cause confusion in IDE class hints etc.. I'm fine either way, only wanted to understand why overloads are preferred here while it also would have worked with defaults. Thanks.